cme/protocols/wmi/wmiexec_regout.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#
+# Author: xiaolichan
+# Link: https://github.com/XiaoliChan/wmiexec-RegOut/blob/main/wmiexec-regOut.py
+# Note: windows version under NT6 not working with this command execution way
+#       https://github.com/XiaoliChan/wmiexec-RegOut/blob/main/wmiexec-reg-sch-UnderNT6-wip.py -- WIP
+# 
+# Description: 
+#   For more details, please check out my repository.
+#   https://github.com/XiaoliChan/wmiexec-RegOut
+
+import sys
+import argparse
+import time
+import logging
+import uuid
+import base64
+
+from impacket.dcerpc.v5.dcomrt import DCOMConnection, COMVERSION
+from impacket.dcerpc.v5.dcom import wmi
+from impacket.dcerpc.v5.dtypes import NULL
+from cme.helpers.logger import highlight
+
+OUTPUT_FILENAME = '__' + str(time.time())
+
+class WMIEXEC_REGOUT:
+    def __init__(self, win32Process, iWbemServices, address, logger):
+        self.logger = logger
+        self.address = address
+        self.__output = '\\' + OUTPUT_FILENAME
+        self.__outputBuffer = str('')
+        self.__shell = 'cmd.exe /Q /c '
+        #self.__pwsh = 'powershell.exe -NoP -NoL -sta -NonI -W Hidden -Exec Bypass -Enc '
+        self.__pwsh = 'powershell.exe -Enc '
+        self.__win32Process = win32Process
+        self.iWbemServices = iWbemServices
+        self.__pwd = str('C:\\')
+
+    def encodeCommand(self, data):
+        data = '$ProgressPreference="SilentlyContinue";' + data
+        data = self.__pwsh + base64.b64encode(data.encode('utf-16le')).decode()
+        return data
+
+    def execute_remote(self, data):
+        # Save result as txt file
+        resultTXT = "C:\\windows\\temp\\" + str(uuid.uuid4()) + ".txt"
+        self.logger.highlight("[+] Executing command: \" %s \""%data)
+        data = data + " > " + resultTXT
+        command = self.__shell + self.encodeCommand(data)
+        self.__win32Process.Create(command, self.__pwd, None)
+        self.logger.highlight("[+] Waiting 5 for command completely executed.")
+        time.sleep(5)
```

Comment:
timeout should be an option that can be configured by the user

Reply:
Sure